src/main/java/org/isf/opd/service/OpdIoOperationRepositoryImpl.java

```diff
@@ -82,17 +82,17 @@ private TypedQuery<Opd> getOpdQuery(
 					cb.equal(opd.join("ward").get("code"), ward.getCode())
 			);
 		}
-		if (diseaseTypeCode != null) {
+		if (diseaseTypeCode != null && !diseaseTypeCode.equals("")) {
 			predicates.add(
 					cb.equal(opd.join("disease").join("diseaseType").get("code"), diseaseTypeCode)
 			);
 		}
-		if (diseaseCode != null) {
+		if (diseaseCode != null && !diseaseCode.equals("")) {
 			predicates.add(
 					cb.equal(opd.join("disease").get("code"), diseaseCode)
 			);
 		}
-		if (ageFrom != 0 || ageTo != 0) {
+		if (ageFrom != 0 && ageTo != 0 ) {
```

Comment:
&& (AND) will cause the interval 0-x (from 0 to x) would not enter in the if block.
Please repleace it with || (OR)